TACC/Core-CMS-Custom
# task/WP-156-Exceptions-Admin-Table-UTH-Requests

## Overview
Adding filters, action dropdown and edit modal for admin exception table

## Related

- [WP-156](https://jira.tacc.utexas.edu/browse/WP-156)


## Changes

- Added status and organization filters and url paths for query params
- Added action dropdown to view and edit exceptions
- To edit an exception, created: 

> 1. Edit exception modal template
> 2. update_exception query
> 3. Modified modal styles for extension
> 4. Success and error templates
> 5. post function in views.py
- Reordered view exception per requests
- Made sure when page is narrowed, the column names align with the new row names


## Testing

1. Go to [http://localhost:8000/submissions/other-exception/](http://localhost:8000/submissions/other-exception/) and submit an exception that will then be stored to the test db.
2. Go to [http://localhost:8000/administration/list-exceptions/?org=All](http://localhost:8000/administration/list-exceptions/)
3. The orgs and statuses are being pulled from the DB directly and not hardcoded in the view. Once the db is updated, it will put all status and orgs available.
- CAREFUL: Try it with the prod DB creds to be sure  the filters are populated correctly. Do not to edit any exception entries that are not tests.
4. Check the org and status filter independently and make sure the entries that display are correct through all pages
5. Try them together and make sure the entries that display are correct through all pages.
6. Try to edit an exception. All fields that you can edit should populate with what is in the DB for that exception_id. 
7. Not all fields are required in the edit modal. Make sure anything you did not change in the edit modal stayed the same after submission.
8. If the exception is a threshold request, the edit modal should allow you to edit the approved threshold % and show the requested threshold %.
9. Check the existing exception information at the bottom of the edit modal. Any field that is not editable at the top should be displayed at the bottom in the same order as the view modal. 
10. Check that the view modal does not have duplicate fields listed

## UI
Exception Admin Table
![Screenshot 2023-06-26 at 5 15 50 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/3973ed05-1c7f-48b5-9114-327d5db94036)


View Exception Modal
![Screenshot 2023-06-26 at 5 16 03 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/edbf6efb-f4b6-47fd-b97d-11e4ae27bf9e)


Status filter  Org filter will only show Tacc Test org on test db
![Screenshot 2023-06-26 at 5 16 22 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/9842d9c5-847d-46df-b8ce-29353d7eb7af)


Edit exception modal pt 1
![Screenshot 2023-06-26 at 5 16 53 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/2827423e-ca68-4753-895a-2b732654f2b3)

If editing a threshold exception, should show threshold field
![Screenshot 2023-06-26 at 5 17 38 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/29204f31-b42d-4862-a1f7-c75c4def8da0)

Edit exception modal pt 2

![Screenshot 2023-06-26 at 5 17 51 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/857d4f6f-d55e-4277-8320-84e7cafe20be)
